From: Bo Jiao <bo.jiao@mediatek.com>
To: Felix Fietkau <nbd@nbd.name>
Subject: [PATCH] mt76: mt7615/mt7915: do reset_work with mt76's work queue
Date: Sat, 7 May 2022 09:21:21 +0800

reset_work may be blocked when mcu message timeout occurs

Signed-off-by: Bo Jiao <Bo.Jiao@mediatek.com>
---
 mt7615/mmio.c | 2 +-
 mt7915/mmio.c | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

--- a/mt7615/mmio.c
+++ b/mt7615/mmio.c
@@ -145,7 +145,7 @@ static void mt7615_irq_tasklet(struct ta
 		return;
 
 	dev->reset_state = mcu_int;
-	ieee80211_queue_work(mt76_hw(dev), &dev->reset_work);
+	queue_work(dev->mt76.wq, &dev->reset_work);
 	wake_up(&dev->reset_wait);
 }
 
--- a/mt7915/mmio.c
+++ b/mt7915/mmio.c
@@ -612,7 +612,7 @@ static void mt7915_irq_tasklet(struct ta
 		mt76_wr(dev, MT_MCU_CMD, val);
 		if (val & MT_MCU_CMD_ERROR_MASK) {
 			dev->reset_state = val;
-			ieee80211_queue_work(mt76_hw(dev), &dev->reset_work);
+			queue_work(dev->mt76.wq, &dev->reset_work);
 			wake_up(&dev->reset_wait);
 		}
 	}
